TeXmacs/progs/convert/markdown/markdownout.scm

```diff
@@ -0,0 +1,483 @@
+  (with-global num-line-breaks 0
+    (with-global indent ""
+      (with-global paragraph-width #f
+        (postlude-add (cdr x))
+        (string-append "[^" (number->string footnote-nr) "]")))))
+
+(define (md-doc-title x)
+  (set! doc-title (md-string (serialize-markdown (cdr x))))
+  (if (hugo-extensions?) ""
+      ((md-header 1) (cdr x))))
+
+(define (md-block x)
+  (with-global num-line-breaks 1
+    (with syntax (tm-ref x 0)
+      (string-concatenate 
+       `("```" ,syntax "\n" ,@(map serialize-markdown (cddr x)) "```\n")))))
+
+(define (md-hugo-tags x)
+  (if (hugo-extensions?)
+      (begin (set! post-tags (cdr x)) "")
+      (string-append "Tags: " (list->csv (cdr x)))))
+
+(define (md-hugo-shortcode x)
+  (if (hugo-extensions?)
+      (string-concatenate 
+       `("{{< " ,(cadr x) " " ,@(list-intersperse (cddr x) " ") " >}}"))
+      ""))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; dispatch
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define serialize-hash (make-ahash-table))
+(map (lambda (l) (apply (cut ahash-set! serialize-hash <> <>) l)) 
+     (list (list 'strong md-style)
+           (list 'em md-style)
+           (list 'tt md-style)
+           (list 'strike md-style)
+           (list 'block md-block)
+           (list 'document md-document)
+           (list 'quotation md-quotation)
+           (list 'definition md-environment)
+           (list 'definition* md-environment*)           
+           (list 'conjecture md-environment)
+           (list 'conjecture* md-environment*)           
+           (list 'question md-environment)
+           (list 'question* md-environment*)           
+           (list 'algorithm md-environment)
+           (list 'algorithm* md-environment*)           
+           (list 'problem md-environment)
+           (list 'problem* md-environment*)           
+           (list 'theorem md-environment)
+           (list 'theorem* md-environment*)           
+           (list 'proposition md-environment)
+           (list 'proposition* md-environment*)           
+           (list 'corollary md-environment)
+           (list 'corollary* md-environment*)           
+           (list 'lemma md-environment)
+           (list 'lemma* md-environment*)           
+           (list 'proof md-environment)
+           (list 'dueto md-dueto)
+           (list 'math md-math)
+           (list 'equation md-equation)
+           (list 'equation* md-equation)
+           (list 'concat md-concat)
+           (list 'itemize md-list)
+           (list 'enumerate md-list)
+           (list 'enumerate-alpha md-list)
+           (list 'h1 (md-header 1))
+           (list 'h2 (md-header 2))
+           (list 'h3 (md-header 3))
+           (list 'h4 (md-header 4))
+           (list 'doc-title md-doc-title)
+           (list 'doc-running-author md-doc-running-author)
+           (list 'author-name author-add)
```

Comment:
Here's where the texmacs tag `author-name` is mapped to `author-add`. This would need some generalization / reading from preferences.